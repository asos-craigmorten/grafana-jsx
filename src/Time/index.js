import PropTypes from "prop-types";

const Time = ({ from = "now-6h", to = "now" }) => <time from={from} to={to} />;

Time.PropTypes = {
  from: PropTypes.string,
  to: PropTypes.string,
};

export default Time;
