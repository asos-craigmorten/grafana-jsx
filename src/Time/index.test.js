import Time from "./";

const withAllProps = {
  from: "test-from",
  to: "test-to",
};

describe("Time", () => {
  describe("when used in isolation", () => {
    describe("when passed no props", () => {
      it("should render the default time object", () => {
        expect(<Time />).toMatchSnapshot();
      });
    });

    describe("when passed all props", () => {
      it("should render a custom time object", () => {
        expect(<Time {...withAllProps} />).toMatchSnapshot();
      });
    });
  });

  describe("when used within another object", () => {
    describe("when passed no props", () => {
      it("should add a default time object to the outer object", () => {
        expect(
          <test>
            <Time />
          </test>
        ).toMatchSnapshot();
      });
    });

    describe("when passed all props", () => {
      it("should add a custom time object to the outer object", () => {
        expect(
          <test>
            <Time {...withAllProps} />
          </test>
        ).toMatchSnapshot();
      });
    });
  });
});
