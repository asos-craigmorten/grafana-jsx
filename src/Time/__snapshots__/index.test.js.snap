// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time when used in isolation when passed all props should render a custom time object 1`] = `
Object {
  "from": "test-from",
  "to": "test-to",
}
`;

exports[`Time when used in isolation when passed no props should render the default time object 1`] = `
Object {
  "from": "now-6h",
  "to": "now",
}
`;

exports[`Time when used within another object when passed all props should add a custom time object to the outer object 1`] = `
Object {
  "time": Object {
    "from": "test-from",
    "to": "test-to",
  },
}
`;

exports[`Time when used within another object when passed no props should add a default time object to the outer object 1`] = `
Object {
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
}
`;
